django/db/models/fields/__init__.py

```diff
@@ -2019,10 +2033,10 @@ def rel_db_type(self, connection):
         if connection.features.related_fields_match_type:
             return self.db_type(connection)
         else:
-            return IntegerField().db_type(connection=connection)
+            return self.integer_field_class().db_type(connection=connection)
```

Comment:
This will always return the same as `self.db_type()` which is not intended. I added `__mro__[1:]` to fix this.